### PR TITLE
Flow docs enhancement: add examples for Grafana Cloud

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
#### PR Description

Flow docs enhancement: add metrics, logs, and traces examples for Grafana Cloud.

#### Which issue(s) this PR fixes

None

#### PR Checklist

- [ ] CHANGELOG updated
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
Thanks @rfratto . 
Very valid point. I'll align with @richardqlam and @gouthamve and get back to you ASAP.

**Comment by cyrille-leclerc** [comment]
@Eve832 can we please discuss the best solution to help Grafana Agents users who connect to Grafana Cloud set their configuration?

**Comment by marctc** [comment]
Maybe @clayton-cornell want's to take a look at this?

**Comment by gouthamve** [comment]
Might be outdated with https://grafana.com/docs/agent/next/flow/getting-started/opentelemetry-to-lgtm-stack/

**Comment by cyrille-leclerc** [comment]
@gouthamve this PR is outdated, for Grafana Cloud, we preferred `GRAFANA_CLOUD_API_KEY`... rather than `TEMPO_PASSWORD`, `LOKI_PASSWORD`...
https://grafana.com/docs/agent/next/flow/getting-started/opentelemetry-to-lgtm-stack/

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 1]
```diff
@@ -320,3 +323,20 @@ prometheus.scrape "demo" {
   forward_to = [prometheus.remote_write.staging.receiver]
 }
 ```
+
+
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:327]:
```suggestion

```

Remove the extra blank line

**Review by clayton-cornell — COMMENTED** [review]
The examples look good to me. 

One general comment. We might want to add an H3 heading for each example... something like:

```
##Examples

The following examples show you how to create `loki.write` components that send received entries to different destinations.

### Send entries to a local Loki instance

<example>

### Send entries to Grafana Cloud

<example>
```

This adds nav elements to the right side of the page, and makes it clear there are two distinct examples.

**Comment by rfratto** [comment]
I think we still have to identify if the open source documentation section is the best place to put documentation specific to Grafana Cloud.

**Comment by clayton-cornell** [comment]
@rfratto I've been discussing this with the team. Generally we want to have a clear division between Grafana Cloud and OSS, but... the Grafana Cloud example here looks useful, and it's not the only solution or example provided here. Could we keep it and provide a description that says something along the lines of:

```
### Send entries to a hosted endpoint

You can create a `loki.write` component that sends your data to a hosted endpoint, for example, Grafana Cloud.

<example>
```

**Comment by rfratto** [comment]
> @rfratto I've been discussing this with the team. Generally we want to have a clear division between Grafana Cloud and OSS, but... the Grafana Cloud example here looks useful, and it's not the only solution or example provided here. Could we keep it and provide a description that says something along the lines of:
> 
> ```
> ### Send entries to a hosted endpoint
> 
> You can create a `loki.write` component that sends your data to a hosted endpoint, for example, Grafana Cloud.
> 
> <example>
> ```

I think that's a good middle ground 👍

File: docs/sources/flow/reference/components/loki.write.md [hunk 2]
```diff
@@ -141,7 +141,7 @@ information.
 * `loki_write_batch_retries_total` (counter): Number of times batches have had to be retried.
 * `loki_write_stream_lag_seconds` (gauge): Difference between current time and last batch timestamp for successful sends.
 
-## Example
+## Examples
 
 This example creates a `loki.write` component that sends received entries to a
 local Loki instance:
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.write.md:147]:
```suggestion

The following examples show you how to create `loki.write` components that send log entries to different destinations.

### Send log entries to a local Loki instance

You can create a `loki.write` component that sends your log entries to a local Loki instance:
```

File: docs/sources/flow/reference/components/loki.write.md [hunk 3]
```diff
@@ -153,3 +153,17 @@ loki.write "local" {
     }
 }
 ```
+
+This example creates a `loki.write` component that sends received entries to Grafana Cloud
+with username and password injected through environment variables:
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.write.md:158]:
```suggestion
### Send log entries to a hosted endpoint

You can create a `loki.write` component that sends your log entries to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 4]
```diff
@@ -155,7 +155,7 @@ configuration.
 `otelcol.exporter.otlp` does not expose any component-specific debug
 information.
 
-## Example
+## Examples
 
 This example creates an exporter to send data to a locally running Grafana
 Tempo without TLS:
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:161]:
```suggestion
The following examples show you how to create an exporter to send data to different destinations.

### Send data to a local Tempo instance

You can create an exporter that sends your data to a local Grafana Tempo instance without TLS:
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 5]
```diff
@@ -171,3 +171,20 @@ otelcol.exporter.otlp "tempo" {
     }
 }
 ```
+
+This example creates an exporter to send data to Grafana Cloud with username and password 
+injected through environment variables:
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:176]:
```suggestion
### Send data to a hosted endpoint

You can create an exporter that sends your data to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
```

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 6]
```diff
@@ -295,7 +295,10 @@ information.
 * `prometheus_remote_storage_exemplars_in_total` (counter): Exemplars read into
   remote storage.
 
-## Example
+## Examples
+
+This example creates a `prometheus.remote_write` component that sends received
+metrics to a local Mimir instance:
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:301]:
```suggestion
The following examples show you how to create `prometheus.remote_write` components that send metrics to different destinations.

### Send metrics to a local Mimir instance

You can create a `prometheus.remote_write` component that sends your metrics to a local Mimir instance:
```

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 7]
```diff
@@ -320,3 +323,20 @@ prometheus.scrape "demo" {
   forward_to = [prometheus.remote_write.staging.receiver]
 }
 ```
+
+
+This example creates a `prometheus.remote_write` component that sends received
+metrics to Grafana Cloud with username and password injected through environment
+variables:
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:330]:
```suggestion
### Send metrics to a hosted endpoint

You can create a `prometheus.remote_write` component that sends your metrics to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
```

**Review by clayton-cornell — COMMENTED** [review]
Added some descriptive text and headings to the examples

File: docs/sources/flow/reference/components/loki.write.md [hunk 8]
```diff
@@ -153,3 +157,18 @@ loki.write "local" {
     }
 }
 ```
+
+### Send log entries to a hosted endpoint
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/loki.write.md:161]:
```suggestion
### Send log entries to Grafana Cloud
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 9]
```diff
@@ -171,3 +174,21 @@ otelcol.exporter.otlp "tempo" {
     }
 }
 ```
+
+### Send data to a hosted endpoint
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:178]:
```suggestion
### Send data to Grafana Cloud
```

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 10]
```diff
@@ -320,3 +326,19 @@ prometheus.scrape "demo" {
   forward_to = [prometheus.remote_write.staging.receiver]
 }
 ```
+
+### Send metrics to a hosted endpoint
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:330]:
```suggestion
### Send metrics to Grafana Cloud
```

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 11]
```diff
@@ -320,3 +326,19 @@ prometheus.scrape "demo" {
...
+
+You can create a `prometheus.remote_write` component that sends your metrics to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
+
+```river
+prometheus.remote_write "default" {
+  endpoint {
+    url = "https://prometheus-xxx.grafana.net/api/prom/push"
+      basic_auth {
+        username = env("PROMETHEUS_USERNAME")
+        password = env("PROMETHEUS_PASSWORD")
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:340]:
```suggestion
        password = env("GRAFANA_CLOUD_API_KEY")
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 12]
```diff
@@ -171,3 +174,21 @@ otelcol.exporter.otlp "tempo" {
...
+otelcol.exporter.otlp "grafana_cloud_tempo" {
+    client {
+        endpoint = "https://tempo-xxx.grafana.net/tempo"
+        auth     = otelcol.auth.basic.grafana_cloud_tempo.handler
+    }
+}
+
+otelcol.auth.basic "grafana_cloud_tempo" {
+    username = env("TEMPO_USERNAME")
+    password = env("TEMPO_PASSWORD")
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:192]:
```suggestion
    password = env("GRAFANA_CLOUD_API_KEY")
```

File: docs/sources/flow/reference/components/loki.write.md [hunk 13]
```diff
@@ -153,3 +157,18 @@ loki.write "local" {
...
+
+You can create a `loki.write` component that sends your log entries to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
+
+```river
+loki.write "default" {
+    endpoint {
+        url = "https://logs-xxx.grafana.net"
+        basic_auth {
+            username = env("LOKI_USERNAME")
+            password = env("LOKI_PASSWORD")
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/loki.write.md:171]:
```suggestion
            password = env("GRAFANA_CLOUD_API_KEY")
```

**Comment by cyrille-leclerc** [comment]
As we use an example with a Grafana Cloud hostname (`.grafana.net`), I'm interested in doing a clean Grafana Cloud example: calling it "Grafana Cloud", referring to "Grafana Cloud API Key" rather than "mimir password / tempo password / loki password"...
 
The concept of "hosted instance" is unclear to me, in my previous companies, "hosted" was a referencing to "self-hosted / self managed".

Then for the local instance, I'm not clear why our example is not secured with authentication and encryption, is it because our OSS versions don't support authentication & encryption? I see examples without auth & encryption as recommendation to not secure your instances.

**Comment by clayton-cornell** [comment]
> The concept of "hosted instance" is unclear to me

We could rephrase to "Cloud instance" or some similar term. The intent I had was to make it a little clearer that we could sent to almost any cloud instance, not just Grafana Cloud, and that Grafana Cloud was being used as the example destination.

**Comment by cyrille-leclerc** [comment]
>> The concept of "hosted instance" is unclear to me

> We could rephrase to "Cloud instance" or some similar term. The intent I had was to make it a little clearer that we could sent to almost any cloud instance, not just Grafana Cloud, and that Grafana Cloud was being used as the example destination.

As the example contains `grafana.net` hostname and `GRAFANA_CLOUD_API_KEY`, I would stick to "Grafana Cloud".

In addition, I could imagine renaming "Local instance" to "Unsecured development instance" and then add a production grade example.

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 14]
```diff
@@ -320,3 +326,19 @@ prometheus.scrape "demo" {
   forward_to = [prometheus.remote_write.staging.receiver]
 }
 ```
+
+### Send metrics to Grafana Cloud
+
+You can create a `prometheus.remote_write` component that sends your metrics to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:332]:
```suggestion
You can create a `prometheus.remote_write` component that sends your metrics to Grafana Cloud. The Prometheus username and the Grafana Cloud API Key are injected in this example through environment variables.
```

File: docs/sources/flow/reference/components/loki.write.md [hunk 15]
```diff
@@ -153,3 +157,18 @@ loki.write "local" {
     }
 }
 ```
+
+### Send log entries to Grafana Cloud
+
+You can create a `loki.write` component that sends your log entries to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/loki.write.md:163]:
```suggestion
You can create a `loki.write` component that sends your log entries Grafana Cloud. The Loki username and Grafana Cloud API Key are injected in this example through environment variables.
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 16]
```diff
@@ -171,3 +174,21 @@ otelcol.exporter.otlp "tempo" {
     }
 }
 ```
+
+### Send data to Grafana Cloud
+
+You can create an exporter that sends your data to a hosted endpoint, for example Grafana Cloud. The username and password are injected through environment variables.
```
Comment (cyrille-leclerc) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:180]:
```suggestion
You can create an `otlp` exporter that sends your data to Grafana Cloud. The Tempo username and Grafana Cloud API Key are injected in this example through environment variables.
```

**Comment by rfratto** [comment]
If the header is going to explicitly call out Grafana Cloud, I'm back to being uncomfortable with this PR. I'm OK with Grafana Cloud being mentioned as an example cloud provider, but not in the header, as it seems more exclusionary towards other vendors.

**Review by clayton-cornell — COMMENTED** [review]
Forgot to finish the review :-( Suggested a change that might meet everyone's expectations. Then we can get this merged?

File: docs/sources/flow/reference/components/loki.write.md [hunk 17]
```diff
@@ -153,3 +157,18 @@ loki.write "local" {
     }
 }
 ```
+
+### Send log entries to Grafana Cloud
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.write.md:161]:
```suggestion
### Send log entries to a managed service
```

File: docs/sources/flow/reference/components/loki.write.md [hunk 18]
```diff
@@ -153,3 +157,18 @@ loki.write "local" {
...
+
+You can create a `loki.write` component that sends your log entries Grafana Cloud. The Loki username and Grafana Cloud API Key are injected in this example through environment variables.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.write.md:163]:
```suggestion
You can create a `loki.write` component that sends your log entries to a managed service, for example, Grafana Cloud.. The Loki username and Grafana Cloud API Key are injected in this example through environment variables.
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 19]
```diff
@@ -171,3 +174,21 @@ otelcol.exporter.otlp "tempo" {
     }
 }
 ```
+
+### Send data to managed service
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:178]:
```suggestion
### Send data to a managed service
```

**Comment by clayton-cornell** [comment]
Closing  this PR. The changes have been moved into: https://github.com/grafana/agent/pull/4799  The merge conflicts were taking too long to resolve. It was simply faster/easier to open a new PR and link back to this one for history.

**Pull request closed without merging**